### PR TITLE
[FLINK-4813][flink-test-utils] make the hadoop-minikdc dependency optional

### PR DESCRIPTION
**Pull request by NicoK**
This removes the need to add the `maven-bundle-plugin`plugin for most projects using `flink-test-utils`.

Instead, any project using `flink-test-utils` that also requires
`SecureTestEnvironment` must add a dependency to `hadoop-minikdc` itself, e.g. in
`pom.xml`:

```xml
   ...
   <dependencies>
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-minikdc</artifactId>
       <version>${minikdc.version}</version>
       <scope>compile</scope>
     </dependency>
   ...
   </dependencies>
   ...

   <build>
     <plugins>
       <!--
         https://issues.apache.org/jira/browse/DIRSHARED-134
         Required to pull the Mini-KDC transitive dependency
       -->
       <plugin>
       <groupId>org.apache.felix</groupId>
       <artifactId>maven-bundle-plugin</artifactId>
       <version>3.0.1</version>
       <inherited>true</inherited>
       <extensions>true</extensions>
     </plugin>
   ...
```

### CODE REVIEW

**Comment by StephanEwen** [comment]
Very good, I had stumbled over that problem as well.

Could you also remove the bundle plugin from the root pom (this should speed up dependency resolutions in IDEs by a lot) and instead add the above described dependency to the test projects that use the `SecureTestEnvironment`?

**Comment by NicoK** [comment]
sure, that makes sense
actually, I only had to add it to the flink-test-utils sub-project since all the others already included the bundler :)

File: flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/SecureTestEnvironment.java [hunk 1]
```diff
@@ -37,9 +37,39 @@
...
  * Helper {@link SecureTestEnvironment} to handle MiniKDC lifecycle.
  * This class can be used to start/stop MiniKDC and create secure configurations for MiniDFSCluster
- * and MiniYarn
+ * and MiniYarn.
+ *
+ * If you use this class in your project, please make sure to add a dependency to
+ * <tt>hadoop-minikdc</tt>, e.g. in your <tt>pom.xml</tt>:
+ * <pre>
+ * ...
+ * &lt;dependencies>
```
Comment (StephanEwen) [inline flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/SecureTestEnvironment.java]: If you do the example code as follows, you can use the `<` and make the sample more readable.
```
<pre>{@code

<dependency>
    <groupId>org.apache.hadoop</groupId>
    ...
</dependency>

}</pre>
```

**Comment by StephanEwen** [comment]
The change looks good, thank you!
Merging this...

**Pull request closed without merging**